src/main/kotlin/world/World.kt

```diff
@@ -1,31 +1,55 @@
 package com.dlfsystems.world
 
 import com.dlfsystems.compiler.Compiler
+import com.dlfsystems.value.VString
 import com.dlfsystems.value.Value
 import com.dlfsystems.vm.Context
 import com.dlfsystems.world.trait.SysTrait
 import com.dlfsystems.world.trait.Trait
-import kotlin.uuid.Uuid
+import com.dlfsystems.world.trait.UserTrait
 import kotlinx.serialization.Serializable
+import ulid.ULID
 
 @Serializable
 data class World(
     val name: String = "world"
 ) {
 
-    val traits: MutableMap<Uuid, Trait> = mutableMapOf()
-    val traitIDs: MutableMap<String, Uuid> = mutableMapOf()
+    val traits: MutableMap<ULID, Trait> = mutableMapOf()
+    val traitIDs: MutableMap<String, ULID> = mutableMapOf()
 
-    val objs: MutableMap<Uuid, Obj> = mutableMapOf()
+    val objs: MutableMap<ULID, Obj> = mutableMapOf()
+
+    fun getUserLogin(name: String, password: String): Obj? {
+        val c = Context(this)
+        getTrait("user")?.objects?.forEach { obj ->
+            objs[obj]?.getProp(c, "username")?.also {
+                if (it == VString(name)) {
+                    objs[obj]?.getProp(c, "password")?.also {
+                        if (it == VString(password)) {
+                            return objs[obj]
+                        }
+                    }
+                }
+            }
+        }
+        return null
+    }
 
     fun getTrait(named: String) = traits[traitIDs[named]]
-    fun getTrait(id: Uuid) = traits[id]
+    fun getTrait(id: ULID) = traits[id]
+    fun getObj(id: ULID) = objs[id]
 
     fun getSysValue(c: Context, name: String): Value = getTrait("sys")!!.getProp(c, name)!!
 
     fun addTrait(name: String): Trait {
         if (traits.values.none { it.name == name }) {
-            return (if (name == "sys") SysTrait() else Trait(name)).also {
+            return when (name) {
+                "sys" -> SysTrait()
+                "user" -> UserTrait()
+                else -> Trait(name)
+            }.also {
+                it.world = this
```

Comment:
There will only ever be one world, right?  If so, you might consider adding a companion object for World that holds a static ref to the current world.  Then you could just reference it as `World.current` anywhere that it wasn't already handy.